File: Java Project/Iter#1/src/EnrollmentRequest.java
package src;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;

public class EnrollmentRequest {

    private ArrayList<Course> courses;
    private Student student;
    private HashMap<String, Schedule> schedule;
    private HashMap<String, String> result;
    private int creditLimit;

    public EnrollmentRequest(ArrayList<Course> courses, Student student, HashMap<String, Schedule> schedule) {
        this.courses = courses;
        this.student = student;
        this.schedule = schedule;
    }

    public ArrayList<Course> getCourses() {
        return courses;
    }

    public void setCourses(ArrayList<Course> courses) {
        this.courses = courses;
    }

    public HashMap<String, String> getResult() {
        return result;
    }

    public void setResult(HashMap<String, String> result) {
        this.result = result;
    }

    public Student getStudent() {
        return student;
    }

    public void setStudent(Student student) {
        this.student = student;
    }

    public HashMap<String, Schedule> getSchedule() {
        return schedule;
    }

    public void setSchedule(HashMap<String, Schedule> schedule) {
        this.schedule = schedule;
    }

    public void appendResult(HashMap<String, String> result) {
        this.result.putAll(result);
    }

    public int getCreditLimit() {
        return creditLimit;
    }

    public void setCreditLimit(int creditLimit) {
        this.creditLimit = creditLimit;
    }

    public int getTotalCredit() {
        int credit = 0;
        for (Course course : this.courses) {
            credit += course.getCredit();
        }
        return credit;
    }

    /* public void checkDuplicate(HashMap hashMap){
         Map<ArrayList<Schedule>, ArrayList<Object>> reverseMap = new HashMap<>();


         for (Object schedules: hashMap.values()) {
             if (!reverseMap.containsKey(entry.getValue())) {
                 reverseMap.put(entry.getValue(), new ArrayList<>());
             }
             ArrayList<Object> keys = reverseMap.get(entry.getValue());
             keys.add(entry.getKey());
             reverseMap.put(entry.getValue(), keys);
         }


     }*/
    public HashMap<String, String> checkDuplicate(HashMap<String, Schedule> hashMap) {
        HashMap<String, String> duplicate = new HashMap<>();
        for (Map.Entry<String, Schedule> entry : hashMap.entrySet()) {
            for (Map.Entry<String, Schedule> entry2 : hashMap.entrySet()) {
                if (entry.getValue().equals(entry2.getValue()) && !entry.getKey().equals(entry2.getKey())) {
                    duplicate.put(entry.getKey(), entry2.getKey());
                }
            }
        }
        return duplicate;
    }
}

File: Python/src/EnrollmentRequest.py
class EnrollmentRequest:
    def __init__(self, courses, student):
        self.__courses = courses
        self.__student = student
        self.__schedule = {}
        self.__result = {}
        self.__credit_limit = 0
        self.__error = ""

    def get_courses(self):
        return self.__courses

    def set_courses(self, courses):
        self.__courses = courses

    def get_result(self):
        return self.__result

    def set_result(self, result):
        self.__result = result

    def get_student(self):
        return self.__student

    def set_student(self, student):
        self.__student = student

    def get_schedule(self):
        return self.__schedule

    def set_schedule(self, schedule):
        self.__schedule = schedule

    def append_result(self, result1):
        temp = self.__result
        temp.update(result1)
        self.set_result(temp)

    def get_credit_limit(self):
        return self.__credit_limit

    def set_credit_limit(self, credit_limit):
        self.__credit_limit = credit_limit

    def get_total_credit(self):
        credit = 0
        for course in self.__courses:
            credit += course.get_credit()
        return credit

    def get_error(self):
        return self.__error

    def set_error(self, error):
        self.__error = error

    def __str__(self):
        return "EnrollmentRequest{{courses={}, student={}, schedule={}, result={}, credit_limit={}, error='{}'}}".format(self.__courses, self.__student, self.__schedule, self.__result, self.__credit_limit, self.__error)